provider/core_route_table_resource.go

```diff
@@ -198,6 +222,15 @@ func (s *RouteTableResourceCrud) SetData() {
 }
 
 func (s *RouteTableResourceCrud) Delete() (e error) {
+	if s.D.Get("default_id") != "" {
+		// We can't actually delete a default resource.
+		// Instead, remove all the route rules first and mark it as deleted.
```

Comment:
This is a good point. Should the fake delete operation reset the resource to its default state? I'll have to think this over a bit more. Is there any reason this is done for route rules but not for security lists and DHCP options?

Reply:
For route rules, it's necessary to clear them to avoid dangling references to gateways. If we don't clear them, then "terraform destroy" will fake delete the route table and then try to delete the gateway. This would fail because the route rules are still referring to the gateway.

I'm leaning towards resetting the resource to its default state. For security list and DHCP options, it's a little more complicated because the default settings rely on values from the parent VCN resource.

One way we could handle this is to read and store the default state of the resource on Create, before we apply an updated settings. Then when we need to reset, just apply the stored state of the resource.